### PR TITLE
Consolidate NFT reward listings

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

Uses `RewardList` from create flow in both NFT listings in reconfig (add/edit a collection)

## Screenshots or screen recordings

Screen Shot 2023-03-07 at 5 16 57 pm

Screen Shot 2023-03-07 at 5 17 13 pm

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/Create/components/RewardsList/RewardItem.tsx [hunk 1]
```diff
@@ -54,6 +56,8 @@ export const RewardItem = ({
   const { data: contentType } = useContentType(fileUrl)
   const isVideo = fileTypeIsVideo(contentType)
 
+  const hasBeneficiary = Boolean(beneficiary) && beneficiary !== AddressZero
```
Comment (tomquirk) [inline src/components/Create/components/RewardsList/RewardItem.tsx:59]: **suggestion**: can use isZeroAddress: https://github.com/jbx-protocol/juice-interface/blob/584bd7ab27a64733f563cb552868061fe1b11f5a/src/utils/address.ts#L30

File: src/components/Create/components/RewardsList/RewardItem.tsx [hunk 2]
```diff
@@ -101,7 +105,7 @@ export const RewardItem = ({
               title={t`Minimum contribution`}
               stat={`${numberUpToPrecisionFormat(contributionFloor)} ETH`}
             />
-            {!!maxSupply && (
+            {!!maxSupply && maxSupply !== DEFAULT_NFT_MAX_SUPPLY && (
```
Comment (tomquirk) [inline src/components/Create/components/RewardsList/RewardItem.tsx:108]: **suggestion** perhaps we should have a single util for hasUnlimitedSpply or something

File: src/components/Create/components/RewardsList/RewardsList.tsx [hunk 3]
```diff
@@ -72,25 +75,23 @@ export const RewardsList: React.FC<RewardsListProps> &
...
-                      removeReward(reward.id)
-                    }}
-                  />
-                  {shouldRenderNftDivider(i) ? (
-                    <Divider className="m-0 mt-12" />
-                  ) : null}
-                </div>
-              ))}
+            {sortNftsByContributionFloor(rewards).map((reward, i) => (
+              <div key={reward.id}>
```
Comment (tomquirk) [inline src/components/Create/components/RewardsList/RewardsList.tsx:79]: **question** do we need the div?

**Review by tomquirk — APPROVED** [review]
@johnnyd-eth ACE. Great work here. I'll assume you've tested all paths (create flow, edit existing NFTs, add new NFTs) so will approve. Just left a few comments.

**Comment by johnnyd-eth** [comment]
> @johnnyd-eth ACE. Great work here. I'll assume you've tested all paths (create flow, edit existing NFTs, add new NFTs) so will approve. Just left a few comments.

Noticed today that the `EditingNfts` hook wasn't working with `RewardsList`. I've tested now and all 3 paths are working well. I thought you might want to have a quick look over my changes to `useEditingNfts` since I added a decent sized diff, so have re-requested your review.

File: src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/EditNftsPage/EditNftsSection.tsx [hunk 4]
```diff
@@ -38,7 +43,10 @@ export function EditNftsSection() {
         {rewardTiers && rewardTiers.length > 0 && (
           <RewardsList
             value={rewardTiers}
-            onChange={setRewardTiers}
+            onChange={newRewardTiers => {
+              deriveAndSetEditedIds(newRewardTiers)
+              setRewardTiers(newRewardTiers)
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/EditNftsPage/EditNftsSection.tsx:48]: @johnnyd-eth i dont thinkn these should be separate calls. They should happen in one call.

File: src/components/Create/components/RewardsList/RewardsList.tsx [hunk 5]
```diff
@@ -76,7 +76,7 @@ export const RewardsList: React.FC<RewardsListProps> &
         {!!rewards.length && (
           <>
             {sortNftsByContributionFloor(rewards).map((reward, i) => (
-              <div key={reward.id}>
+              <>
                 <RewardItem
                   reward={reward}
```
Comment (tomquirk) [inline src/components/Create/components/RewardsList/RewardsList.tsx:81]: @johnnyd-eth we actually do need the div. Or else we cant set the `key` prop and react will skitz out

Comment (johnnyd-eth) [reply src/components/Create/components/RewardsList/RewardsList.tsx:81]: Ah yep got it

File: src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/EditNftsPage/EditNftsSection.tsx (hunk 4 above)
Comment (johnnyd-eth) [reply src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/EditNftsPage/EditNftsSection.tsx:49]: Yep good call, changed it up.

**Pull request merged**